plugins/module_utils/foreman_helper.py

```diff
@@ -691,12 +923,16 @@ def exit_json(self, **kwargs):
             kwargs['entity'] = self._after_full
         super(ForemanEntityAnsibleModule, self).exit_json(**kwargs)
 
+    @property
+    def blacklisted_fields(self):
+        return [key for key, value in self.entity_spec.items() if value.get('no_log', False)]
+
 
 class ForemanTaxonomicEntityAnsibleModule(ForemanEntityAnsibleModule):
     def __init__(self, argument_spec=None, **kwargs):
         spec = dict(
             organizations=dict(type='entity_list', flat_name='organization_ids'),
-            locations=dict(type='entity_list', flat_name='location_ids'),
+            locations=dict(type='entity_list', flat_name='location_ids', search_by='title'),
```

Comment:
I wonder if we could/should have a list of entities that should be `search_by=title` by default (locations, hostgroups, operatingsystems come to mind).

Or would that be too  much magic?

Reply:
I was thinking about this too, a list of entities that we don't search by name.